### PR TITLE
Barbara frame adjustment

### PR DESCRIPTION
**Pull request by skippi**
Frame fixes based off the sheet in #209 .

### CODE REVIEW

File: internal/characters/barbara/attack.go [hunk 1]
```diff
@@ -10,17 +11,56 @@ import (
...
+	attackFrames[1] = frames.InitNormalCancelSlice(attackHitmarks[1], 28)
+	attackFrames[1][action.ActionAttack] = 21
+	attackFrames[1][action.ActionCharge] = 24
+
+	// N3 -> x
+	attackFrames[2] = frames.InitNormalCancelSlice(attackHitmarks[2], 30)
+	attackFrames[2][action.ActionAttack] = 22
+	attackFrames[2][action.ActionCharge] = 28
+
+	// N3 -> x
```
Comment (k0l11) [inline internal/characters/barbara/attack.go:41]: should say N4 in the comment here

File: internal/characters/barbara/burst.go [hunk 2]
```diff
@@ -10,27 +10,34 @@ import (
 var burstFrames []int
 
 func init() {
-	burstFrames = frames.InitAbilSlice(110)
+	burstFrames = frames.InitAbilSlice(198)
+	burstFrames[action.ActionDash] = 161
+	burstFrames[action.ActionJump] = 162
+	burstFrames[action.ActionSwap] = 198
```
Comment (k0l11) [inline internal/characters/barbara/burst.go:16]: this line is redundant

File: internal/characters/barbara/burst.go [hunk 3]
```diff
@@ -10,27 +10,34 @@ import (
...
-	c.ConsumeEnergy(8)
-	c.SetCD(action.ActionBurst, 20*60)
+	c.ConsumeEnergy(6)
+	c.SetCDWithDelay(action.ActionBurst, 20*60, 1)
 
 	return action.ActionInfo{
 		Frames:          frames.NewAbilFunc(burstFrames),
 		AnimationLength: burstFrames[action.InvalidAction],
-		CanQueueAfter:   burstFrames[action.InvalidAction],
+		CanQueueAfter:   burstFrames[action.ActionSkill],
```
Comment (k0l11) [inline internal/characters/barbara/burst.go:40]: shortest cancel is Charge not Skill

File: internal/characters/barbara/charge.go [hunk 4]
```diff
@@ -10,14 +10,20 @@ import (
...
 
 func init() {
-	chargeFrames = frames.InitAbilSlice(90)
+	chargeFrames = frames.InitAbilSlice(89)
+	chargeFrames[action.ActionDash] = 56
+	chargeFrames[action.ActionJump] = 56
+	chargeFrames[action.ActionSwap] = 57
+	chargeFrames[action.ActionSkill] = 88
+	chargeFrames[action.ActionBurst] = 87
+	chargeFrames[action.ActionAttack] = 89
```
Comment (k0l11) [inline internal/characters/barbara/charge.go:22]: this line is redundant

File: internal/characters/barbara/attack.go [hunk 5]
```diff
@@ -10,17 +11,56 @@ import (
...
+	attackFrames[3] = frames.InitNormalCancelSlice(attackHitmarks[3], 60)
+	attackFrames[3][action.ActionWalk] = 57
+	attackFrames[3][action.ActionDash] = 2
+	attackFrames[3][action.ActionJump] = 3
+	attackFrames[3][action.ActionSwap] = 5
+	attackFrames[3][action.ActionAttack] = 60
```
Comment (k0l11) [inline internal/characters/barbara/attack.go:47]: this line is redundant

**Review by shizukayuki — COMMENTED** [review]
the `x -> swap` cancels are slightly off since it was counted until `model swap` not `swap icon ungray`. and i don't know how i feel about N4 being special case here.

other than the comments below everything else look fine to me.

File: internal/characters/barbara/attack.go [hunk 6]
```diff
@@ -51,22 +90,35 @@ func (c *char) Attack(p map[string]int) action.ActionInfo {
 			})
 			done = true
 		}
+	}
 
+	adjustedFrames := attackFrames
+	adjustedHitmarks := attackHitmarks
+	currState := c.Core.Player.CurrentState()
+	if currState == action.DashState || (currState == action.NormalAttackState && c.NormalCounter == 0) {
```
[the remark this answers is not in the document]
Comment (skippi) [reply internal/characters/barbara/attack.go:98]: Discussed partly on discord, keeping since this is a thing: https://www.youtube.com/watch?v=66CQ3c1dwzw

**Pull request merged**